testing/common_testing_util.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+import pytest
+from ivadomed.scripts import download_data as ivadomed_download_data
+import os
+import shutil
+import sys
+
+# Cannot use __ivadomed_dir__ here as that would refer to the IvadoMed installed in the Conda/Venv
+# Absolute path to the temporary folder, which should sit at the root of the repo.
+# We could also replace this hard coded levels reference with a simple loop to search for .git/.github sitting at the
+# root of the repo. To be discussed.
+FOLDER_LEVELS_FROM_ROOT = 1
+
+path_repo_root: Path = Path(__file__).parents[FOLDER_LEVELS_FROM_ROOT].absolute()
+
+path_temp: str = str(path_repo_root / "tmp")
```

Comment:
I feel it might be worth having a discussion about how and when we could introduce typing into the ivado codebase. Perhaps we could make a milestone related to adding type hinting throughout the code?

Reply:
I do remember discussing this about a month ago in a meeting and people seemed to be onboard with adding types to the code, but I don't think we ever drafted a specific plan to introduce it

Reply:
Yeah, great point of context which I did not mention in the PR earlier. A proposal has been put together but I do not believe it has been effectively shared/communicated to the everyone as it is still under discussion. You can see the preliminary proposal [here](https://docs.google.com/document/d/1wgKjqU1lGQanjXZxYzjjwp8vdA_QiJgQ88B5efjzT1w/edit?usp=sharing) on the gradual adoption of type hinting going forward and methods to help reinforce it across various projects at NeuroPoly. Feedbacks welcome!

That is why for all new code I am adding (especially key ones like the common_util, I aim to ensure they are being increasingly incorporated well).